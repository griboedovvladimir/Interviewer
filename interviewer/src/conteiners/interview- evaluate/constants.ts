export const FORM_ID = 'evaluteForm';
export const BUTTON_TEXT_EVALUTE = 'evalute';
export const BUTTON_TEXT_EDIT = 'edit';
export const BUTTON_EVALUTE_CLASS = 'mdc-component__containers__primary';
export const BUTTON_EDIT_CLASS = 'mdc-component__containers__secondary';
export const SNACKBAR_CLASS_HIDDEN = 'snackbar-hidden';
export const SNACKBAR_CLASS_ACTIVE = 'snackbar-active';