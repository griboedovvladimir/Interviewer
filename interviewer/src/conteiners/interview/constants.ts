export const ACTION_GET_QUESTION = 'QUESTION';
export const ACTION_GET_INTERVIEW = 'GET_INTERVIEW';