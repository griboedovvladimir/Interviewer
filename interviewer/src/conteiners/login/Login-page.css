.login-form {
  width: 300px;
}
.form-wrapper {
  display: flex;
  justify-content: center;
  padding-top: 10%;
}
.login-form {
  width: 300px;
}
.form-content {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  padding: 15px;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12);
}
