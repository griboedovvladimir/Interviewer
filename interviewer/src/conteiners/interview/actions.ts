import * as CONSTANTS from "./constants";

export function getQuestion(question: any) {
    return {
        type: CONSTANTS.ACTION_GET_QUESTION,
        question
    };
}

export function getInterview(interviews: any) {
    return {
        type: CONSTANTS.ACTION_GET_INTERVIEW,
        interviews
    };
}