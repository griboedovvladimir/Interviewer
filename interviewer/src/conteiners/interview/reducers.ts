import * as CONSTANTS from './constants';


export function question(state = {
    currentQuestionNumber: 1,
    raiting: '',
    text: '',
    source: '',
    question_id: 0,
    subtopic_id: 0
}, action: any) {
    if (action.type === CONSTANTS.ACTION_GET_QUESTION) {
        return {...action.question}
    }
    return state;
}