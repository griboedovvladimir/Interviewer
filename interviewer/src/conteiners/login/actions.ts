import * as CONSTANTS from "./constants"

export function login(logged: boolean) {
    return {
        type: CONSTANTS.ACTION_LOGIN,
        logged
    };
}