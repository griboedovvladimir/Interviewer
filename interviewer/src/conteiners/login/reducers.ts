import * as CONSTANTS from "./constants";
import * as ROOT_CONSTANTS from "../../constants";

export function login(state = {logged: !!localStorage.getItem(ROOT_CONSTANTS.LOCAL_STORAGE_KEY_AUTH_KEY)}, action: any) {
    if (action.type === CONSTANTS.ACTION_LOGIN) {
        return {
            logged: action.logged,
        };

    }
    if (action.type === CONSTANTS.ACTION_LOGOUT) {
        return {
            logged: action.logged,
        };

    }
    return state;
}