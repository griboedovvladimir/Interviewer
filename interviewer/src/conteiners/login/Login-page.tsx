import * as React from "react";
import {Component} from 'react';
import {bindActionCreators, Dispatch} from 'redux';
import {Redirect} from "react-router";
import {connect} from 'react-redux';
import * as CONSTANTS from '../../constants';
import * as actions from './actions';
import {CheckLoginService} from '../../services/check-login.service';
import {AuthorizationService} from "../../services/authorization.service";
import bound from '../../decorators/bound'


class LoginPage extends Component {
    public props: any;
    public authorization:AuthorizationService;

    constructor(props: any, private service: CheckLoginService) {
        super(props);
        this.service = new CheckLoginService;
        this.authorization = new AuthorizationService();
    }

    @bound public handleSubmit(e: any) {
        e.preventDefault();
        this.service.checkUser(e.target.loginName.value, e.target.loginPassword.value)
            .then(res => res.json().then(check => {
                if (check.authorization === CONSTANTS.LOGGED_API_RES) {
                    this.props.action.login(true);
                    this.authorization.authorizate(check.token);
                }
            }));
    }

    public render() {
        if (!this.props.logged) {
            return (
                <div className="form-wrapper">
                    <form className="login-form" onSubmit={this.handleSubmit}>
                        <div className="mdc-component mdc-component__textfield">
                            <div dir-horizontal="" className="mdc-component__containers">
                                <div className="mdc-component__section mdc-component__section--size-narrow">
                                    <div className="mdc-component__containers__primary mdc-textfield--theme-primary">
                                        <div className="mdc-textfield">
                                            <input type="text" id="loginName" className="mdc-textfield__input"
                                                   placeholder="Your name" name="loginName" pattern="[A-z.]+"
                                                   required={true}
                                                   minLength={3}/>
                                            <label htmlFor="loginName"
                                                   className="mdc-textfield__label mdc-textfield__label--float-above">Name</label>
                                        </div>
                                    </div>
                                    <div className="mdc-component__containers__secondary mdc-textfield--theme-primary">
                                        <div className="mdc-textfield">
                                            <input type="password" id="loginPassword" className="mdc-textfield__input"
                                                   name="loginPassword" placeholder="Your password" required={true}/>
                                            <label htmlFor="loginPassword"
                                                   className="mdc-textfield__label mdc-textfield__label--float-above">Passsword</label>
                                        </div>
                                    </div>
                                    <div className="mdc-component mdc-component__buttons">
                                        <div className="mdc-component__containers__primary">
                                            <button type="submit" className="mdc-button mdc-button--raised">Submit
                                            </button>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </form>
                </div>
            );
        } else {
            return <Redirect to={CONSTANTS.MAIN_PAGE}/>;
        }
    }
}


const mapStateToProps = (state: any) => ({
    logged: state.login.logged,
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
    action: bindActionCreators({...actions}, dispatch)
});


export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);