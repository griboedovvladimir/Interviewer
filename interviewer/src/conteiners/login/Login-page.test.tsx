import * as React from 'react';
import { shallow } from 'enzyme';
import * as actions from './actions';
import * as CONSTANTS from './constants';
import {login} from "./reducers";

const logged = true;
/*-------------------------- Redux tests -------------------------------*/
describe('login actions', () => {
    it('should change boolean value logged', () => {
        const expectedAction = {
            type: CONSTANTS.ACTION_LOGIN,
            logged
        };
        expect(actions.login(logged)).toEqual(expectedAction)
    })
});

describe('login reducers', () => {
    it('should handle ACTION_LOGIN', () => {
        expect(
            login({logged: false}, {
                type: CONSTANTS.ACTION_LOGIN,
                logged: true
            })
        ).toEqual({logged: true});

        expect(
            login({logged: false}, {
                type: CONSTANTS.ACTION_LOGIN,
                logged: true
            })
        ).toEqual({logged: true});

        expect(
            login(undefined, {
                type: CONSTANTS.ACTION_LOGIN,
                logged: true
            })
        ).toEqual({logged: true});

        expect(
            login({logged: false}, {
                type: '',
                logged: true
            })
        ).toEqual({logged: false})

    });

    it('should handle ACTION_LOGOUT', () => {
        expect(
            login({logged: false}, {
                type: CONSTANTS.ACTION_LOGOUT,
                logged: true
            })
        ).toEqual({logged: true});

        expect(
            login({logged: false}, {
                type: CONSTANTS.ACTION_LOGIN,
                logged: true
            })
        ).toEqual({logged: true});

        expect(
            login({logged: true}, {
                type: '',
                logged: false
            })
        ).toEqual({logged: true})
    });


});
/*----------------------- Components tests----------------------------*/
import {LoginPage} from './Login-page';


describe("<LoginPage />  UI Component", ()=>{
    it("render default login-page", ()=>{
        shallow(<LoginPage/>)
            .find('div.form-wrapper')
            .length
    });
});

    it('should render LoginPage in preview mode', () => {
        const wrapper = shallow(
            <LoginPage/>
        );

        expect(wrapper).toMatchSnapshot();

        wrapper.find('[name="login-btn"]').simulate('click');

        expect(wrapper).toMatchSnapshot();
    });


    describe('when submiting the form', () => {
        const wrapper = shallow(
            <LoginPage/>
        );
        expect(wrapper).toMatchSnapshot();
        beforeEach(() => {
            wrapper.find('form').simulate('submit', {
                preventDefault: () => {},
            })
        });
    });

    describe('when clicking the form', () => {
        const wrapper = shallow(
            <LoginPage/>
        );
        expect(wrapper).toMatchSnapshot();
        beforeEach(() => {
            wrapper.find('[name="login-btn"]').simulate('click', {
                preventDefault: () => {},
            })
        });
    });


