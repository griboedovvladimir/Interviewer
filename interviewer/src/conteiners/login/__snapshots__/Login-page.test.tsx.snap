// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="form-wrapper"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="mdc-component mdc-component__textfield"
    >
      <div
        className="mdc-component__containers"
        dir-horizontal=""
      >
        <div
          className="mdc-component__section mdc-component__section--size-narrow"
        >
          <div
            className="mdc-component__containers__primary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginName"
                minLength={3}
                name="loginName"
                pattern="[A-z.]+"
                placeholder="Your name"
                required={true}
                type="text"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginName"
              >
                Name
              </label>
            </div>
          </div>
          <div
            className="mdc-component__containers__secondary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginPassword"
                name="loginPassword"
                placeholder="Your password"
                required={true}
                type="password"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginPassword"
              >
                Passsword
              </label>
            </div>
          </div>
          <div
            className="mdc-component mdc-component__buttons"
          >
            <div
              className="mdc-component__containers__primary"
            >
              <button
                className="mdc-button mdc-button--raised"
                name="login-btn"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[` 2`] = `
<div
  className="form-wrapper"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="mdc-component mdc-component__textfield"
    >
      <div
        className="mdc-component__containers"
        dir-horizontal=""
      >
        <div
          className="mdc-component__section mdc-component__section--size-narrow"
        >
          <div
            className="mdc-component__containers__primary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginName"
                minLength={3}
                name="loginName"
                pattern="[A-z.]+"
                placeholder="Your name"
                required={true}
                type="text"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginName"
              >
                Name
              </label>
            </div>
          </div>
          <div
            className="mdc-component__containers__secondary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginPassword"
                name="loginPassword"
                placeholder="Your password"
                required={true}
                type="password"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginPassword"
              >
                Passsword
              </label>
            </div>
          </div>
          <div
            className="mdc-component mdc-component__buttons"
          >
            <div
              className="mdc-component__containers__primary"
            >
              <button
                className="mdc-button mdc-button--raised"
                name="login-btn"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[` 3`] = `
<div
  className="form-wrapper"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="mdc-component mdc-component__textfield"
    >
      <div
        className="mdc-component__containers"
        dir-horizontal=""
      >
        <div
          className="mdc-component__section mdc-component__section--size-narrow"
        >
          <div
            className="mdc-component__containers__primary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginName"
                minLength={3}
                name="loginName"
                pattern="[A-z.]+"
                placeholder="Your name"
                required={true}
                type="text"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginName"
              >
                Name
              </label>
            </div>
          </div>
          <div
            className="mdc-component__containers__secondary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginPassword"
                name="loginPassword"
                placeholder="Your password"
                required={true}
                type="password"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginPassword"
              >
                Passsword
              </label>
            </div>
          </div>
          <div
            className="mdc-component mdc-component__buttons"
          >
            <div
              className="mdc-component__containers__primary"
            >
              <button
                className="mdc-button mdc-button--raised"
                name="login-btn"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[` 4`] = `
<div
  className="form-wrapper"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="mdc-component mdc-component__textfield"
    >
      <div
        className="mdc-component__containers"
        dir-horizontal=""
      >
        <div
          className="mdc-component__section mdc-component__section--size-narrow"
        >
          <div
            className="mdc-component__containers__primary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginName"
                minLength={3}
                name="loginName"
                pattern="[A-z.]+"
                placeholder="Your name"
                required={true}
                type="text"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginName"
              >
                Name
              </label>
            </div>
          </div>
          <div
            className="mdc-component__containers__secondary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginPassword"
                name="loginPassword"
                placeholder="Your password"
                required={true}
                type="password"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginPassword"
              >
                Passsword
              </label>
            </div>
          </div>
          <div
            className="mdc-component mdc-component__buttons"
          >
            <div
              className="mdc-component__containers__primary"
            >
              <button
                className="mdc-button mdc-button--raised"
                name="login-btn"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`<LoginPage />  UI Component should render LoginPage in preview mode 1`] = `
<div
  className="form-wrapper"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="mdc-component mdc-component__textfield"
    >
      <div
        className="mdc-component__containers"
        dir-horizontal=""
      >
        <div
          className="mdc-component__section mdc-component__section--size-narrow"
        >
          <div
            className="mdc-component__containers__primary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginName"
                minLength={3}
                name="loginName"
                pattern="[A-z.]+"
                placeholder="Your name"
                required={true}
                type="text"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginName"
              >
                Name
              </label>
            </div>
          </div>
          <div
            className="mdc-component__containers__secondary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginPassword"
                name="loginPassword"
                placeholder="Your password"
                required={true}
                type="password"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginPassword"
              >
                Passsword
              </label>
            </div>
          </div>
          <div
            className="mdc-component mdc-component__buttons"
          >
            <div
              className="mdc-component__containers__primary"
            >
              <button
                className="mdc-button mdc-button--raised"
                name="login-btn"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`<LoginPage />  UI Component should render LoginPage in preview mode 2`] = `
<div
  className="form-wrapper"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="mdc-component mdc-component__textfield"
    >
      <div
        className="mdc-component__containers"
        dir-horizontal=""
      >
        <div
          className="mdc-component__section mdc-component__section--size-narrow"
        >
          <div
            className="mdc-component__containers__primary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginName"
                minLength={3}
                name="loginName"
                pattern="[A-z.]+"
                placeholder="Your name"
                required={true}
                type="text"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginName"
              >
                Name
              </label>
            </div>
          </div>
          <div
            className="mdc-component__containers__secondary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginPassword"
                name="loginPassword"
                placeholder="Your password"
                required={true}
                type="password"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginPassword"
              >
                Passsword
              </label>
            </div>
          </div>
          <div
            className="mdc-component mdc-component__buttons"
          >
            <div
              className="mdc-component__containers__primary"
            >
              <button
                className="mdc-button mdc-button--raised"
                name="login-btn"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`<LoginPage />  UI Component should render Markdown in preview mode 1`] = `
<div
  className="form-wrapper"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="mdc-component mdc-component__textfield"
    >
      <div
        className="mdc-component__containers"
        dir-horizontal=""
      >
        <div
          className="mdc-component__section mdc-component__section--size-narrow"
        >
          <div
            className="mdc-component__containers__primary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginName"
                minLength={3}
                name="loginName"
                pattern="[A-z.]+"
                placeholder="Your name"
                required={true}
                type="text"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginName"
              >
                Name
              </label>
            </div>
          </div>
          <div
            className="mdc-component__containers__secondary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginPassword"
                name="loginPassword"
                placeholder="Your password"
                required={true}
                type="password"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginPassword"
              >
                Passsword
              </label>
            </div>
          </div>
          <div
            className="mdc-component mdc-component__buttons"
          >
            <div
              className="mdc-component__containers__primary"
            >
              <button
                className="mdc-button mdc-button--raised"
                name="login-btn"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`<LoginPage />  UI Component should render Markdown in preview mode 2`] = `
<div
  className="form-wrapper"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="mdc-component mdc-component__textfield"
    >
      <div
        className="mdc-component__containers"
        dir-horizontal=""
      >
        <div
          className="mdc-component__section mdc-component__section--size-narrow"
        >
          <div
            className="mdc-component__containers__primary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginName"
                minLength={3}
                name="loginName"
                pattern="[A-z.]+"
                placeholder="Your name"
                required={true}
                type="text"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginName"
              >
                Name
              </label>
            </div>
          </div>
          <div
            className="mdc-component__containers__secondary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginPassword"
                name="loginPassword"
                placeholder="Your password"
                required={true}
                type="password"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginPassword"
              >
                Passsword
              </label>
            </div>
          </div>
          <div
            className="mdc-component mdc-component__buttons"
          >
            <div
              className="mdc-component__containers__primary"
            >
              <button
                className="mdc-button mdc-button--raised"
                name="login-btn"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`should render LoginPage in preview mode 1`] = `
<div
  className="form-wrapper"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="mdc-component mdc-component__textfield"
    >
      <div
        className="mdc-component__containers"
        dir-horizontal=""
      >
        <div
          className="mdc-component__section mdc-component__section--size-narrow"
        >
          <div
            className="mdc-component__containers__primary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginName"
                minLength={3}
                name="loginName"
                pattern="[A-z.]+"
                placeholder="Your name"
                required={true}
                type="text"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginName"
              >
                Name
              </label>
            </div>
          </div>
          <div
            className="mdc-component__containers__secondary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginPassword"
                name="loginPassword"
                placeholder="Your password"
                required={true}
                type="password"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginPassword"
              >
                Passsword
              </label>
            </div>
          </div>
          <div
            className="mdc-component mdc-component__buttons"
          >
            <div
              className="mdc-component__containers__primary"
            >
              <button
                className="mdc-button mdc-button--raised"
                name="login-btn"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`should render LoginPage in preview mode 2`] = `
<div
  className="form-wrapper"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="mdc-component mdc-component__textfield"
    >
      <div
        className="mdc-component__containers"
        dir-horizontal=""
      >
        <div
          className="mdc-component__section mdc-component__section--size-narrow"
        >
          <div
            className="mdc-component__containers__primary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginName"
                minLength={3}
                name="loginName"
                pattern="[A-z.]+"
                placeholder="Your name"
                required={true}
                type="text"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginName"
              >
                Name
              </label>
            </div>
          </div>
          <div
            className="mdc-component__containers__secondary mdc-textfield--theme-primary"
          >
            <div
              className="mdc-textfield"
            >
              <input
                className="mdc-textfield__input"
                id="loginPassword"
                name="loginPassword"
                placeholder="Your password"
                required={true}
                type="password"
              />
              <label
                className="mdc-textfield__label mdc-textfield__label--float-above"
                htmlFor="loginPassword"
              >
                Passsword
              </label>
            </div>
          </div>
          <div
            className="mdc-component mdc-component__buttons"
          >
            <div
              className="mdc-component__containers__primary"
            >
              <button
                className="mdc-button mdc-button--raised"
                name="login-btn"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
